storybookjs/storybook
# Load Angular library projects

Issue:

## What I did

Angular 6 has the ability to generate new library projects using the `ng generate library my-lib` command.

This will by default create projects in the the `projects/` folder.

Currently the storybook angular cli does not consult these locations.

This pull request adds support for these libraries.

It does so by:
 - loading angular.json.
 - collecting all projects which are defined as being a library.
-  loading the ng-package.json file of the library to determine the module's entry point (normally src/public_api.ts)
- loading the package.json of the library to determine module name.
- creating webpack aliases for the libraries so the modules can be referenced from anywhere.

In order to also glob the stories in `projects/` the user has to manually add this to `./storybook/config.js`

Hot reload works out of the box and didn't need any code changes.

## How to test

Checkout this branch and run storybook, the `test-lib` library appears under Projects.
If you modify any of the project files the story should reload and reflect the changes.

N.B. The test-lib project was generated with the cli, I didn't make any code changes on it, beside naming the selector to 'storybook-test-lib' while this was required by the lint rules.
